src/averageAges.js

```diff
@@ -56,6 +81,21 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  let child = people.filter(children =>
+    people.some(mother =>
+      children.mother === mother.name));
+  const son = child.filter((peoples) => peoples.sex === 'm');
+
+  child = onlyWithSon === true
+    ? son
+    : child;
+
+  const ages = child.map((kid) =>
+    kid.born - people.find(women => women.name === kid.mother).born);
```

Comment:
agree it looks way much readable
```suggestion
  const ageDifferencesSum = child.reduce((sum, children) => {
    const mother = people.find(person => children.mother === person.name);
    const ageDifference = children.born - mother.born;
      
    return sum + ageDifference;
  });
  
  const averageAgeDifference = ageDifferencesSum / children.length;
```